### PR TITLE
DIV-6509: Add ServiceApplicationNotApproved

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DIV-6509

### Change description ###

Add with Permissions:

|CaseEvent|Admin|Beta|La|Sol|Super|Cit
-- |-- | --| -- |--|--|--
|aosNotReceivedFromServiceAppRejected| R | CRU | R | R | R | - 
|solAosNotReceivedFromServiceAppRejected| R | R | R | CRU | R | - 
|startAosFromServiceAppRejected| R | CRU | R | R | R | CRU 
|answerReceivedFromServiceAppRejected| R | CRU | R | R | R | R 
|coRespAnswerReceivedFromServiceAppRejected| R | CRU | R | R | R | R 
|IssueAOSOffline_Respondent_FromServiceAppRejected| R | CRU | R | R | R | R 
|IssueAOSOffline_CoRespondent_FromServiceAppRejected| R | CRU | R | R | R | R 
|reissueFromServiceAppRejected| R | CRU | R | R | R | R 
|transferCaseFromServiceAppRejected| R | CRU | R | R | R | R 

ACs |  
-- | --
1 | "CallBackURLSubmittedEvent":   "${CCD_DEF_COS_URL}/petition-updated",
2 | "CallBackURLSubmittedEvent":   "${CCD_DEF_COS_URL}/petition-updated",
3 | "CallBackURLSubmittedEvent":   "${CCD_DEF_COS_URL}/petition-updated",
4 * | "CallBackURLSubmittedEvent":   "${CCD_DEF_COS_URL}/petition-updated",
5 * | "CallBackURLAboutToSubmitEvent":   "${CCD_DEF_COS_URL}/co-respondent-answered",
6 * | "CallBackURLAboutToSubmitEvent":   "${CCD_DEF_COS_URL}/issue-aos-pack-offline/parties/respondent",
7 * | "CallBackURLAboutToSubmitEvent":   "${CCD_DEF_COS_URL}/issue-aos-pack-offline/parties/co-respondent",
8 | "CallBackURLAboutToSubmitEvent":   "${CCD_DEF_COS_URL}/petition-issued?generateAosInvitation=true",

# Testing Criteria
|*Testing Criteria*|*Pass/Fail*|Comment|
-- | -- | ---
|As a caseworker, If the case is in the state 'ServiceApplicationNotApproved', and the petitioner is NOT represented then the event AOS Not Received within SLA can be seen from the dropdown| ✅  |N/A|
|As a caseworker, If the case is in the state 'ServiceApplicationNotApproved', and the petitioner is represented then the event AOS Not Received within SLA can be seen from the dropdown| ✅ |N/A|
|In Swagger, As a respondent, If a case is in the state 'ServiceApplicationNotApproved' and 'pin' and 'caseID' (triggering the new event 'startAosFromServiceAppRejected' then the case state should change to 'AOSStarted'| |N/A|
|In Swagger, If a case is in the state 'ServiceApplicationNotApproved', and answers received from Respondent (using the new event 'answerReceivedFromServiceAppRejected' then case is moved into state 'Defend divorce'| |N/A|
|in Swagger, If a case is in the state 'ServiceApplicationNotApproved', and answers received from Co-respondent (using the new event 'coRespAnswerReceivedFromServiceAppRejected' then case stays in the state 'ServiceApplicationNotApproved'| |N/A|
|As a caseworker, If the case is in the state 'ServiceApplicationNotApproved' then the event 'Issue AOS pack to respondent' can be seen from the dropdown| ✅ |N/A|
|As a caseworker, If the case is in the state 'ServiceApplicationNotApproved' then the event 'Issue AOS pack to Co-respondent' can be seen from the dropdown| ✅ |N/A|
|As a caseworker, If the case is in the state 'ServiceApplicationNotApproved' then the event 'Reissue petition' can be seen from the dropdown| ✅ |N/A|
|As a caseworker, If the case is in the state 'ServiceApplicationNotApproved' then the event 'Transfer from CTSC to RDC', can be seen from the dropdown| ✅ |N/A|

### CODE REVIEW

**Review by yarpo — APPROVED** [review]

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (Dizrupto) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:229]: added this event as part of https://github.com/hmcts/div-ccd-definitions/pull/488, I think you would just need to rebase as it's been merged to master.

Comment (yarpo) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:229]: yes, there is code conflict.

```
Conflicting files
definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
```

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (Dizrupto) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:283]: As mentioned above, you can remove this one as it already exists in the same file and also the authorisations you had added for it :)

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (Dizrupto) [inline definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:260]: I'm not sure hmm, did you do a merge with master or something like this? Seems you have each event twice :). Maybe a merging bug

File: definitions/divorce/json/CaseEvent/CaseEvent.json
Comment (Dizrupto) [inline definitions/divorce/json/CaseEvent/CaseEvent.json:71]: Is that part of this ticket?

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (jacmurphy) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:260]: Yes, it was a merging bug.
I had rebased, it made sense. I tried to push, conflict, pull and then it essentially merged with master... undoing it all now

**Review by yarpo — CHANGES_REQUESTED** [review]
please move definitions from `core` files to nonprod files, properly feature toggled.

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent.json:1270]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseField/AuthorisationCaseField.json:614]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:169]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

Comment (yarpo) [reply definitions/divorce/json/CaseEvent/CaseEvent-deemed-and-dispensed-nonprod.json:169]: Ok, I have removed duplicates.

Please don't use rebase. Just merge code into your branch:

```
git pull origin master
```

to get the latest master. We squash everything before we merge it into master so it makes little sense to rebase your branch over and over again.

And as you can see you don't know what code ends up in the final version of your branch. Rebase is tricky. I dopn;t use it if I don't have to.

File: definitions/divorce/json/CaseEventToFields/CaseEventToFields.json
Comment (yarpo) [inline definitions/divorce/json/CaseEventToFields/CaseEventToFields.json:936]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: definitions/divorce/json/CaseField/CaseField.json
Comment (yarpo) [inline definitions/divorce/json/CaseField/CaseField.json:948]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: definitions/divorce/json/CaseTypeTab/CaseTypeTab.json
Comment (yarpo) [inline definitions/divorce/json/CaseTypeTab/CaseTypeTab.json:3388]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: definitions/divorce/json/ComplexTypes/ComplexTypes.json
Comment (yarpo) [inline definitions/divorce/json/ComplexTypes/ComplexTypes.json:209]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: definitions/divorce/json/FixedLists/FixedLists.json
Comment (yarpo) [inline definitions/divorce/json/FixedLists/FixedLists.json:1270]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: test/unit/definitions/divorce/CaseTypeTab.test.js
Comment (yarpo) [inline test/unit/definitions/divorce/CaseTypeTab.test.js:95]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

File: test/unit/definitions/divorce/CaseTypeTab.test.js
Comment (yarpo) [inline test/unit/definitions/divorce/CaseTypeTab.test.js:138]: I don't know why it's here. It was a part of:
https://github.com/hmcts/div-ccd-definitions/pull/423

already merged into master. Should not be displayed here.

**Review by yarpo — APPROVED** [review]
Some code displayed in diff of this PR come from
https://github.com/hmcts/div-ccd-definitions/pull/423

it's already merged into master. Should not be displayed here.

I don't know why git is displaying it. Please verify what you have is valid.

Maybe lets test it on PR and not merge it into master before testing

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:168]: ok,

```
aosNotReceivedFromServiceAppRejected | R | CRU | R | R | R | -
```

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:222]: solicitor CRU,
R fior others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:159]: beta and citizen CRU
R others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:238]: Beta: CRU

R for others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:280]: beta CRU,
R others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:322]: Beta CRU,
R others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:364]: beta CRU
R others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:406]: beta CRU, R Others

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-deemed-and-dispensed-nonprod.json:448]: beta cru
r others

**Review by yarpo — APPROVED** [review]

**Comment by yarpo** [comment]
the same code was merged here:
https://github.com/hmcts/div-ccd-definitions/pull/491

**Pull request closed without merging**
